packages/tracing/src/transaction.ts

```diff
@@ -113,6 +113,14 @@ export class Transaction extends SpanClass implements TransactionInterface {
       }).endTimestamp;
     }
 
+    // if this transaction (or any of its descendants) is currently set on the scope, pop it off so that no future
+    // errors or spans get associated with it
+    const currentScope = this._hub.getScope();
+    const spanOnScope = currentScope?.getSpan();
+    if (currentScope && spanOnScope?.transaction === this) {
```

Comment:
Is there a case in which `getSpan()` will return the transaction itself, and in that case is it correct to look at `.transaction`?

Reply:
Transactions have a reference to themselves, so it works either way.